include/xgboost/tree_updater.h

```diff
@@ -65,11 +65,13 @@ class TreeUpdater {
     return false;
   }
 
+  virtual char const* Name() const = 0;
```

Comment:
No.  It's only used for debugging.  Keeping it here because I couldn't find a better way to inspect polymorphism.  I need to see that the correct updater is being selected by configuration step.